### ISSUE TITLE
Invalid complex column name handling in ClickHouseBulkCopy

### ISSUE THREAD

**Issue by i-sinister** [DarkWanderer/ClickHouse.Client#54]
I've encountered following problem - when column name requires quotation (contains whitespace, comma, semicolon, bracket, etc) ClickHouseBulkCopy,ClickHouseBulkCopy miserably fails with error like:

> Code: 62, e.displayText() = DB::Exception: Syntax error: failed at position 40 (line 1, col 40): 2) FORMAT RowBinary
> . Expected identifier (version 20.3.20.6 (official build))

Steps to reproduce:
```bash
#!/bin/bash

dotnet new console --name CHBulkInsertStrangeColumnName

dotnet add CHBulkInsertStrangeColumnName/CHBulkInsertStrangeColumnName.csproj  package ClickHouse.Client --version 2.3.0.326

cat << EOF > CHBulkInsertStrangeColumnName/Program.cs
using System;
using System.Threading;
using System.Threading.Tasks;
using ClickHouse.Client.ADO;
using ClickHouse.Client.Copy;

namespace CHBulkInsertStrangeColumnName
{
        internal class Program
        {
                internal static async Task Main()
                {
                        var cancellationToken = CancellationToken.None;
                        Console.Write("Please enter connection string: ");
                        var connectionString = Console.ReadLine();
                        using var connection = new ClickHouseConnection(connectionString);
                        using var command = connection.CreateCommand();
                        command.CommandText = $"DROP TABLE IF EXISTS my_table";
                        await command.ExecuteNonQueryAsync(cancellationToken);
                        command.CommandText = $"CREATE TABLE my_table (column_1 Int64, \"column.2\" Int64) ENGINE = Log";
                        await command.ExecuteNonQueryAsync(cancellationToken);
                        using var bulkCopy =
                                new ClickHouseBulkCopy(connection)
                                {
                                        DestinationTableName = "my_table",
                                        BatchSize = 100000,
                                        MaxDegreeOfParallelism = 1
                                };

                        var rows = new[] { new object[] { 1, 2 } };
                        try
                        {
                                await bulkCopy.WriteToServerAsync(rows, cancellationToken);
                                Console.WriteLine("ClickHouseBulkCopy.WriteToServerAsync completed");
                        }
                        catch (Exception e)
                        {
                                Console.WriteLine(e.Message);
                        }
                }
        }
}
EOF

cd CHBulkInsertStrangeColumnName

dotnet run
```
Script drops and creates table with hardcoded name 'my_table' with a column that contains period in the name. Replacing period with underscore and any letter makes it work but I have to support columns with whitespaces, periods and other characters. 

Using overload that accepts column names does not solves it but make problem more clear:
> Code: 47, e.displayText() = DB::Exception: Missing columns: 'column' while processing query: 'SELECT column_1, column.2 FROM my_table LIMIT 0', required columns: 'column_1' 'column', source columns: 'column.2' 'column_1' (version 20.3.20.6 (official build))

**Comment by DarkWanderer** [comment]
Thank you for a detailed report - should be able to fix that easily by enclosing the columns in backticks

**Comment by DarkWanderer** [comment]
In the meanwhile - can you try explicitly specifying columns in backticks?
```
await bulkCopy.WriteToServerAsync(rows, new[] { "`column.1`, "`column '2'`" });
```

**Comment by DarkWanderer** [comment]
Fixed in [3.0](https://github.com/DarkWanderer/ClickHouse.Client/releases/tag/3.0.0.357)
